### ISSUE TITLE
PanicException: Python API call failed when using map_elements on a column

### ISSUE THREAD

**Issue by elisevansbbfc** [pola-rs/polars#14701]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl
from unidecode import unidecode

title_akas_df = (
    pl.read_csv(
        'data/external/imdb/title_akas.tsv', 
        separator = '\t', 
        dtypes = {'title': str}, 
        ignore_errors = True,
    )
)

title_akas_df['title'].map_elements(unidecode)
```

results in PanicException: Python API call failed

### Log output

```shell
---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 37-38: unexpected end of data
---------------------------------------------------------------------------
PanicException                            Traceback (most recent call last)
Cell In[2], [line 10](vscode-notebook-cell:?execution_count=2&line=10)
      [1](vscode-notebook-cell:?execution_count=2&line=1) title_akas_df = (
      [2](vscode-notebook-cell:?execution_count=2&line=2)     pl.read_csv(
      [3](vscode-notebook-cell:?execution_count=2&line=3)         'data/external/imdb/title_akas.tsv', 
   (...)
      [7](vscode-notebook-cell:?execution_count=2&line=7)     )
      [8](vscode-notebook-cell:?execution_count=2&line=8) )
---> [10](vscode-notebook-cell:?execution_count=2&line=10) title_akas_df['title'].map_elements(unidecode)

File [c:\Users\user\anaconda3\envs\duckdb_env\Lib\site-packages\polars\series\series.py:5329](file:///C:/Users/user/anaconda3/envs/duckdb_env/Lib/site-packages/polars/series/series.py:5329), in Series.map_elements(self, function, return_dtype, skip_nulls)
   [5325](file:///C:/Users/user/anaconda3/envs/duckdb_env/Lib/site-packages/polars/series/series.py:5325)     pl_return_dtype = py_type_to_dtype(return_dtype)
   [5327](file:///C:/Users/user/anaconda3/envs/duckdb_env/Lib/site-packages/polars/series/series.py:5327) warn_on_inefficient_map(function, columns=[self.name], map_target="series")
   [5328](file:///C:/Users/user/anaconda3/envs/duckdb_env/Lib/site-packages/polars/series/series.py:5328) return self._from_pyseries(
-> [5329](file:///C:/Users/user/anaconda3/envs/duckdb_env/Lib/site-packages/polars/series/series.py:5329)     self._s.apply_lambda(function, pl_return_dtype, skip_nulls)
   [5330](file:///C:/Users/user/anaconda3/envs/duckdb_env/Lib/site-packages/polars/series/series.py:5330) )

PanicException: Python API call failed
```

### Issue description

There are letters from different alphabets (e.g. Japanese, Russian, etc.) in the column I am trying to apply the function to, nonetheless they are in the UTF-8 character set.

The function may work on a subset of the data e.g. `title_akas_df['title'][:x].map_elements(unidecode)` for some `x` will work as usual where `x+1` is the first 'problematic' element, but if I then try to access it like `title_akas_df[x]` my kernel crashes.

### Expected behavior

I would expect the code to apply the unidecode function to each string element of the column

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.10
Index type:           UInt32
Platform:             Windows-10-10.0.22631-SP0
Python:               3.11.5 | packaged by Anaconda, Inc. | (main, Sep 11 2023, 13:26:23) [MSC v.1916 64 bit (AMD64)]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          2.2.1
connectorx:           <not installed>
deltalake:            <not installed>
fsspec:               2023.4.0
gevent:               <not installed>
hvplot:               0.8.4
matplotlib:           3.7.2
numpy:                1.24.3
openpyxl:             3.0.10
pandas:               2.0.3
pyarrow:              11.0.0
pydantic:             1.10.8
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           1.4.39
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
```

</details>

**Comment by elisevansbbfc** [comment]
The relevant data can be found [here](https://datasets.imdbws.com/). I am using the `title.akas.tsv.gz` data.

**Comment by mcrumiller** [comment]
FYI the space in the byte array seen above is `\x20`. There is some badly formed unicode in here.

**Comment by mcrumiller** [comment]
Ok, I've find the final reason. This file is not a UTF-8 file, it is a "MS932", "CP932" or "Windows-31J", often called "Shift-JIS". You can see that standard utf8 decoding in python fails:

```python
>>> b = b'\xe3\x82\xa8\xe3\x83\xad\xe4\xba\x8b\xe5\xb8\xab\xe3\x81\x9f\xe3\x81\xa1\xe3\x82\x88\xe3\x82\x8a \xe4\xba\xba\xe9\xa1\x9e\xe5\xad\xa6\xe5\x85\xa5\xe9\x96'
>>> b.decode()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 37-38: unexpected end of data
```
But if we specify `Shift-JIS` it works:
```python
>>> b.decode("Shift-JIS")
'繧ｨ繝ｭ莠句ｸｫ縺溘■繧医ｊ 莠ｺ鬘槫ｭｦ蜈･髢'
```

TL;DR polars doesn't support this encoding, sorry!

**Comment by mcrumiller** [comment]
Also, it's not your fault, the [IMDB Page](https://developer.imdb.com/non-commercial-datasets/) specifically cites Utf-8 as the encoding, but they are just wrong. I'd maybe [contact them](https://help.imdb.com/contact/developer/?ref_=devsite_footer) if you really need this dataset, or you can create a workaround by re-exporting the file with that one record deleted.
